jmix-core/core/src/main/java/io/jmix/core/security/UserManager.java

```diff
@@ -41,13 +42,20 @@ public interface UserManager {
 
 
     /**
-     * Changes the password for the specific user.
+     * Changes the password for the passed users.
      *
-     * @param users - users which need reset passwords
-     * @return map which contains new password for specific user
+     * @param users users which need reset passwords
+     * @return map which contains new password for the passed users
      */
     Map<UserDetails, String> resetPasswords(Set<UserDetails> users);
 
+    /**
+     * Changes the password for a specific user without saving it to the database.
+     *
+     * @param user user which need reset password
+     * @return pair witch contains new password for specific user
+     */
+    Pair<UserDetails, String> resetPasswordWithoutSave(UserDetails user);
```

Comment:
This method is written specifically for your use-case. Let's make it more generic, e.g.:

1. It should accept a set of users, same as `resetPasswords`. 
2. Instead of declaring logic in name, I'd add a parameter:`resetPasswords(Set<UserDetails> users, boolean save)` and "redirect" `resetPasswords(Set)` to this method with default value